src/core/thread/dua_manager.cpp

```diff
@@ -184,6 +249,522 @@ otError DuaManager::Store(void)
+        case ThreadStatusTlv::kDuaDuplicate:
+            RemoveDomainUnicastAddress();
+            mDadCounter++;
+
+            if (GenerateDomainUnicastAddressIid() == OT_ERROR_NONE)
+            {
+                AddDomainUnicastAddress();
+            }
+
+            break;
+        case ThreadStatusTlv::kDuaNoResources:
+            // fall through
+        case ThreadStatusTlv::kDuaNotPrimary:
+            // fall through
+        case ThreadStatusTlv::kDuaGeneralFailure:
+            UpdateReregistrationDelay();
+            break;
+        }
+    }
+    else
+#endif
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    {
+        Child *child = Get<ChildTable>().GetChildAtIndex(mChildIndexDuaRegistering);
+
+        VerifyOrExit(child != NULL, OT_NOOP);
+        VerifyOrExit(child->HasIp6Address(target), OT_NOOP);
+
+        switch (status)
+        {
+        case ThreadStatusTlv::kDuaSuccess:
+            // Mark as Registered
+            mChildDuaRegisterMask.Set(mChildIndexDuaRegistering);
+            break;
+        case ThreadStatusTlv::kDuaReRegister:
+            mRegisterCurrentChildIndex = true;
+            mDelay.mFields.mCheckDelay = Mle::kImmediateReRegisterDelay;
+            break;
+        case ThreadStatusTlv::kDuaInvalid:
+            // fall through
+        case ThreadStatusTlv::kDuaDuplicate:
+            SendAddressNotification(target, static_cast<ThreadStatusTlv::DuaStatus>(status), *child);
+            IgnoreError(child->RemoveIp6Address(target));
+            break;
+        case ThreadStatusTlv::kDuaNoResources:
+            // fall through
+        case ThreadStatusTlv::kDuaNotPrimary:
+            // fall through
+        case ThreadStatusTlv::kDuaGeneralFailure:
+            mChildDuaRegisterMask.Clear(mChildIndexDuaRegistering);
+            UpdateReregistrationDelay();
+
+            break;
+        }
+    }
+#endif // OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+
+exit:
+    ScheduleTimer();
+}
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+void DuaManager::SendAddressNotification(Ip6::Address &             aAddress,
+                                         ThreadStatusTlv::DuaStatus aStatus,
+                                         const Child &              aChild)
+{
+    Coap::Message *  message = nullptr;
+    Ip6::MessageInfo messageInfo;
+    otError          error;
+
+    VerifyOrExit((message = Get<Coap::Coap>().NewPriorityMessage()) != nullptr, error = OT_ERROR_NO_BUFS);
+
+    SuccessOrExit(error =
+                      message->Init(OT_COAP_TYPE_CONFIRMABLE, OT_COAP_CODE_POST, OT_URI_PATH_DUA_REGISTRATION_NOTIFY));
+    SuccessOrExit(error = message->SetPayloadMarker());
+
+    SuccessOrExit(error = Tlv::AppendUint8Tlv(*message, ThreadTlv::kStatus, static_cast<uint8_t>(aStatus)));
+    SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kTarget, &aAddress, sizeof(aAddress)));
+
+    messageInfo.GetPeerAddr().SetToRoutingLocator(Get<Mle::MleRouter>().GetMeshLocalPrefix(),
```

Comment:
just curious: why not use link local address for Child?

Reply:
Good question. For TMF message, I think RLOC should be preferred for compression efficiency: IPv6 prefix compression with context id, and short address in Mac layer.

Does it make sense? 

@jwhui any thoughts from you?

Reply:
For compression, RLOC should allow for smaller packet since it allows short address at MAC header as well.

Reply:
That's what I thought too, Thanks!